jerry-core/mem/mem-context.c

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ * Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mem-context.h"
+
+/** \addtogroup mem Memory allocation
+ * @{
+ *
+ * \addtogroup heap Context
+ * @{
+ */
+
+/**
+ * Global context.
+ */
+jerry_context_t jerry_global_context;
+
+#ifndef JERRY_HEAP_SECTION_ATTR
+
+/**
+ * Global heap.
+ */
+mem_heap_t jerry_global_heap __attribute__ ((aligned (MEM_ALIGNMENT)));
+
+#else /* JERRY_HEAP_SECTION_ATTR */
+
+/**
+ * Jerry global heap section attribute.
+ */
+#define JERRY_GLOBAL_HEAP_SECTION __attribute__ ((section (JERRY_HEAP_SECTION_ATTR)))
+
+/**
+ * Global heap.
+ */
+mem_heap_t jerry_global_heap __attribute__ ((aligned (MEM_ALIGNMENT))) JERRY_GLOBAL_HEAP_SECTION;
+
+#endif /* !JERRY_HEAP_SECTION_ATTR */
```

Comment:
This a is bit overly complex. You could simplify it as:
```c
#ifndef JERRY_HEAP_SECTION_ATTR
#define JERRY_GLOBAL_HEAP_SECTION
#else /* JERRY_HEAP_SECTION_ATTR */
/**
 * Jerry global heap section attribute.
 */
#define JERRY_GLOBAL_HEAP_SECTION __attribute__ ((section (JERRY_HEAP_SECTION_ATTR)))
#endif /* !JERRY_HEAP_SECTION_ATTR */

/**
 * Global heap.
 */
mem_heap_t jerry_global_heap __attribute__ ((aligned (MEM_ALIGNMENT))) JERRY_GLOBAL_HEAP_SECTION;
```
This would move the double definiton of the variable outside the macro guards and remove duplication.